File: backend/src/main/java/com/example/mazerunner/Position.java
package com.example.mazerunner;

public class Position {
    private final int x;
    private final int y;

    public Position(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    @Override
    public boolean equals(Object o) {
        if (o instanceof Position op) {
            return op.x == this.x && op.y == this.y;
        }
        return false;
    }
}

File: frontend/src/app/Position.ts
export class Position {
  x: number;
  y: number;

  constructor(x: number, y: number) {
    this.x = x;
    this.y = y;
  }

  public equals(obj: Object): boolean {
    if (obj instanceof Position)
      return obj.x == this.x && obj.y == this.y;

    return false;
  }
}
